GoogleDataTransport/GDTLibrary/Public/GDTUploadPackage.h

```diff
@@ -16,12 +16,71 @@
 
 #import <Foundation/Foundation.h>
 
+#import <GoogleDataTransport/GDTTargets.h>
+
+@class GDTClock;
 @class GDTStoredEvent;
+@class GDTUploadPackage;
+
+/** A protocol that allows a handler to respond to package lifecycle events. */
+@protocol GDTUploadPackageProtocol <NSObject>
+
+@optional
+
+/** Indicates that the package has expired.
+ *
+ * @note Package expiration will only be checked every 5 seconds.
+ *
+ * @param package The package that has expired.
+ */
+- (void)packageExpired:(GDTUploadPackage *)package;
+
+/** Indicates that the package was successfully delivered.
+ *
+ * @param package The package that was delivered.
+ */
+- (void)packageDelivered:(GDTUploadPackage *)package;
```

Comment:
Why aren't the following two methods one bool method?

Reply:
Good idea